exonum/src/api/private/system.rs

```diff
@@ -162,8 +172,8 @@ impl Api for SystemApi {
 
         let self_ = self.clone();
         let peer_add = move |req: &mut Request| -> IronResult<Response> {
-            let addr: SocketAddr = self_.required_param(req, "ip")?;
-            self_.node_channel.peer_add(addr).map_err(ApiError::from)?;
+            let info: PeerAddInfo = self_.parse(req)?;
```

Comment:
It seems you can destructure here like `let PeerAddInfo { ip } = ...` Would look much more awesome, imo.